docker-app/qfieldcloud/core/tests/test_jobs.py

```diff
@@ -0,0 +1,129 @@
+import logging
+
+from qfieldcloud.authentication.models import AuthToken
+from qfieldcloud.core.models import (
+    Job,
+    PackageJob,
+    Person,
+    ProcessProjectfileJob,
+    Project,
+)
+from rest_framework import status
+from rest_framework.test import APITransactionTestCase
+
+from .utils import (
+    set_subscription,
+    setup_subscription_plans,
+    testdata_path,
+    wait_for_project_ok_status,
+)
+
+logging.disable(logging.CRITICAL)
+
+VALID_LOCALIZED_LAYER_KEY = "apiary_853c29cb_ead8_4932_a60a_5cba33140b3d"
+INVALID_LAYER_KEY = "apiary_e7d1b542_1b1c_422d_ac75_c14d2b54e472"
+INVALID_LOCALIZED_LAYER_KEY = "area_a7b70aff_23f2_4876_9d4e_fa63036df0b2"
+
+
+class QfcTestCase(APITransactionTestCase):
+    def setUp(self):
+        setup_subscription_plans()
+
+        # Create a user
+        self.u1 = Person.objects.create_user(username="u1", password="abc123")
+        self.t1 = AuthToken.objects.get_or_create(user=self.u1)[0]
+
+        # Create a project
+        self.p1 = Project.objects.create(name="p1", is_public=False, owner=self.u1)
+
+    def test_create_job_bad_layer_handler_extracted_values(self):
+        # Test that BadLayerHandler is parsing data properly during package and process projectfile jobs
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.t1.key)
+
+        # Push the QGIS project and datasource GPKG files
+        for local_filename, upload_filename in [
+            ("bumblebees.gpkg", "bumblebees.gpkg"),
+            ("simple_bumblebees_wrong_localized.qgs", "project.qgs"),
+        ]:
+            file_path = testdata_path(local_filename)
+            response = self.client.post(
+                f"/api/v1/files/{self.p1.id}/{upload_filename}/",
+                {
+                    "file": open(file_path, "rb"),
+                },
+                format="multipart",
+            )
+            self.assertTrue(status.is_success(response.status_code))
```

Comment:
```suggestion
        # Push the QGIS project and datasource GPKG files
        for local_filename, upload_filename in [
            ("bumblebees.gpkg", "bumblebees.gpkg"),
            ("simple_bumblebees_wrong_localized.qgs", "project.qgs"),
        ]:
            file_path = testdata_path(local_filename)
            response = self.client.post(
                f"/api/v1/files/{self.p1.id}/{upload_filename}/",
                {
                    "file": open(file_path, "rb"),
                },
                format="multipart",
            )
            self.assertTrue(status.is_success(response.status_code))
```

You can move this to a separate function. Also check for `status.HTTP_201_CREATED` instead of `is_success`.